modules/translation/i18n/i18n.go

```diff
@@ -50,11 +58,69 @@ func (ls *LocaleStore) AddLocaleByIni(langName, langDesc string, localeFile inte
 		UnescapeValueCommentSymbols: true,
 	}, localeFile, otherLocaleFiles...)
 	if err == nil {
-		iniFile.BlockMode = false
-		lc := &locale{store: ls, langName: langName, langDesc: langDesc, messages: iniFile}
-		ls.langNames = append(ls.langNames, lc.langName)
-		ls.langDescs = append(ls.langDescs, lc.langDesc)
-		ls.localeMap[lc.langName] = lc
+		// Common code between production and development.
+		ls.langNames = append(ls.langNames, langName)
+		ls.langDescs = append(ls.langDescs, langDesc)
+
+		// Specify the offset for translationValues.
+		ls.langOffsets = append(ls.langOffsets, len(ls.langOffsets))
+
+		// Distinguish between production and development
+		// For development, live-reload of the translation files is important.
+		// For production, we can do some expensive work and then make the querying fast.
+		if setting.IsProd {
+			// If the language is the default language, then we go through all keys. These keys
+			// will be the only keys that we take into account while querying for a translation
+			if langName == ls.defaultLang {
+				idx := 0
+				// Store all key, value into two slices.
+				for _, section := range iniFile.Sections() {
+					for _, key := range section.Keys() {
+						ls.translationKeys = append(ls.translationKeys, section.Name()+"#"+key.Name())
+						ls.translationValues = append(ls.translationValues, key.Value())
+
+						ls.keyToOffset[strings.TrimPrefix(section.Name()+"."+key.Name(), "DEFAULT.")] = idx
+						idx++
+					}
+				}
```

Comment:
Only `en-US` has full keys (3128 lines at the moment), other translations may have fewer keys. eg: locale_it-IT.ini only have around 2400 lines.

The logic could be:
1. get all keys and translations from `en-US`
2. fill all other translations

Reply:
So instead of depending on defaultLang to get all the keys that we will "support" we hardcode the en-US one?

Reply:
Yup.  I think there are 3 concepts:
1. base language: the language which provides all keys and fallback translations
2. default language: the default language for users who doesn't set their language
3. other languages

At the moment, the base language is en-US, and it's fine to be hard-coded

Reply:
Got it, if nobody objects I'm fine with it.